src/components/QuestionPlay/WithOptions.tsx

```diff
@@ -1,54 +1,100 @@
 import Markdown from '../Markdown';
-import styles from './styles.module.css';
 
 interface Props {
-  setSelectedChoice: Function;
-  selectedOptionId: number | null | string;
+  setSelectedChoices: Function;
+  selectedOptionIds: number[] | null;
   options: IOption[];
-  selectedChoice: number | null | string;
+  selectedChoices: number[] | null;
   isAttempted: boolean;
   isTimerRunning: Boolean;
 }
 
+const useStyles = createStyles((theme) => ({
+  input: {
+    backgroundColor: 'rgb(255, 198, 198)',
+    borderColor: theme.colors.red,
+  },
+}));
+
 export default function WithOptions({
   options,
-  setSelectedChoice,
-  selectedOptionId,
-  selectedChoice,
+  setSelectedChoices,
+  selectedOptionIds,
+  selectedChoices,
   isAttempted,
   isTimerRunning,
 }: Props) {
+  const { classes } = useStyles();
+
+  function getCheckboxColor(option) {
+    if (selectedOptionIds && option.isCorrect && selectedOptionIds.includes(option.optionId)) {
+      return 'green';
+    } else if (
+      selectedOptionIds &&
+      ((!option.isCorrect && selectedOptionIds.includes(option.optionId)) ||
+        (option.isCorrect && !selectedOptionIds.includes(option.optionId)))
+    ) {
+      return 'red';
+    } else {
+      return '';
+    }
+  }
+
   return (
     <>
-      <Checkbox.Group
-        value={selectedChoice === null ? undefined : [selectedChoice.toString()]}
-        orientation="vertical"
-        label={<Title order={6}>Options</Title>}
-        size="md"
-        my="lg"
-        onChange={(value: string[]) => setSelectedChoice(value[value.length - 1])}>
-        {options.map((option) => (
-          <Checkbox
-            key={option.optionId}
-            label={<Markdown>{option.text}</Markdown>}
-            className={classNames({
-              [styles.isAttempted]: isAttempted,
-              [styles.correct]:
-                option.isCorrect && selectedOptionId && parseInt(selectedOptionId.toString()) === option.optionId,
-              [styles.inCorrect]:
-                !option.isCorrect && selectedOptionId && parseInt(selectedOptionId.toString()) === option.optionId,
-            })}
-            value={option.optionId.toString()}
-            disabled={isAttempted || !isTimerRunning}
-            radius="xl"
-            size="lg"
-          />
-        ))}
-      </Checkbox.Group>
+      <Group align="flex-start">
+        {isAttempted && (
+          <Checkbox.Group
+            value={options.filter((x) => x.isCorrect).map((x) => x.optionId.toString())}
+            orientation="vertical"
+            label={<Title order={6}>ANSWER</Title>}
+            size="md"
+            my="lg">
+            {options.map((x) =>
+              x.isCorrect ? (
+                <Checkbox
+                  value={x.optionId.toString()}
+                  checked
+                  color="green"
+                  className="justifyCenter"
+                  style={{ pointerEvents: 'none' }}
```

Comment:
no need to do this if we do disabled=true?

Reply:
disabled checkbox doesn't show the background color, so using pointerEvents: 'none'